### PR TITLE
kubevirt: migrate VmConsoles from web-ui-components

### PR DESCRIPTION
**Pull request by mareklibra**
Part of the gradual effort to remove the kubevirt-web-ui-components dependency.

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx [hunk 1]
```diff
@@ -98,3 +161,29 @@ export const VMConsoleFirehose: React.FC<VMTabProps> = ({ obj: vm }) => {
...
+  vmi?: FirehoseResult<VMIKind>;
+  services?: FirehoseResult<K8sResourceKind[]>;
+  pods?: FirehoseResult<K8sResourceKind[]>;
+  migrations?: FirehoseResult<K8sResourceKind[]>;
+};
+
+type VMConsolesProps = {
+  vm: VMKind;
+  onStartVm: () => void;
+  LoadingComponent: React.FC;
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: React.ComponentType is better

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx [hunk 2]
```diff
@@ -98,3 +161,29 @@ export const VMConsoleFirehose: React.FC<VMTabProps> = ({ obj: vm }) => {
...
+  vmi?: VMIKind;
+  vnc?: any;
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: can you add better types for cnv/serial/rdp ?

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx [hunk 3]
```diff
@@ -98,3 +161,29 @@ export const VMConsoleFirehose: React.FC<VMTabProps> = ({ obj: vm }) => {
...
+  serial?: any;
+  rdp?: any;
+};
+
+type VMIsDownProps = {
+  onStartVm: () => void;
+};
+
+type VMIsStartingProps = {
+  LoadingComponent: React.FC;
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: React.ComponentType

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    </div>
+  );
+};
+DesktopViewerSelector.displayName = 'DesktopViewer';
+
+type DesktopViewerSelectorProps = {
+  vm: VMKind;
+  vmi: VMIKind;
+  guestAgent: boolean;
+  rdpServiceManual: any;
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: can you add better types ? for `vncServiceManual` too

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      const network = networks.find((n) => n.name === i.name);
+      return {
+        name: i.name,
+        type: network.multus ? NetworkType.MULTUS : NetworkType.POD,
+        ip,
+      };
+    });
+};
+
+const getDefaultNetwork = (networks) => {
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: types

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+import './desktop-viewer-selector.scss';
+
+const SELECT_NETWORK_INTERFACE = '--- Select Network Interface ---';
+const NIC = 'Network Interface';
+const GUEST_AGENT_WARNING = 'Guest Agent is not installed on Virtual Machine';
+const GUEST_AGENT_WARNING_TITLE = 'Missing Guest Agent';
+const NO_IP_ADDRESS = 'No IP address is reported for network interface';
+const NO_IP_ADDRESS_TITLE = 'Networks misconfigured';
+
+const getVmRdpNetworks = (vm, vmi) => {
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: types

File: frontend/packages/kubevirt-plugin/src/components/vms/serial-console-connector.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        data,
+        ']',
+      );
+      // data are resent back from backend so _will_ pass through onDataFromBackend
+      ws && ws.send(new Blob([data]));
+    },
+    [ws],
+  );
+
+  const onDisconnect = React.useCallback(() => {
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/serial-console-connector.tsx]: why dont you pass `onBackendDisconnected` directly ?

Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/serial-console-connector.tsx]: sure, left-over after changes ...

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx (hunk 1 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: done

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx (hunk 2 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: done

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx (hunk 3 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx]: done

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx (hunk 4 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: done

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx (hunk 5 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: done

File: frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx (hunk 6 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx]: done

**Pull request merged**